internal/cmd/test/build_cmd.go

```diff
@@ -0,0 +1,112 @@
+package test
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/gobuffalo/envy"
+	"github.com/gobuffalo/meta"
+)
+
+var (
+	// if GO_BIN is set, use it, otherwise use "go"
+	goBinary = envy.Get("GO_BIN", "go")
+
+	// testValuedFlags are the flags on the go test command that
+	// receive a value, we use this to determine if the last argument
+	// is a flag or a package path
+	testValuedFlags = strings.Join([]string{
```

Comment:
Agreed. I don't like this part. I added this to be able to understand when a flag received values, this is important to determine if the test command is being passed packages to test. p.e:

```
go test //
go test -run xxx // No packages
go test -run xxx -p 1 // No packages
go test ./... // packages specified
```

Reply:
Yes, I know this is added to check if the user gave the module list to test or just gave flags (with values).

My idea on this is simple. I think we can just ignore adding an automatically found module list when the user gave additional flags for `go test`. Previously when we need to walk through the given (or automatically collected) modules for `-m` supporting before, with switching between `-testify.m` and `-run`, so the list of target modules was important. However, if we deprecate the behavior of `-m` we had and just use the standard `-run`, we don't need to care about it anymore and we can let the user complete the command line as the same as when the user runs `go test` directly.

Briefly, I would like to make the command the same or similar to the `go test` command line:

* `buffalo test -- -cover ./...` == `go test -cover ./...` _after preparing test db_ (just use the user-provided args as is)
* `buffalo test` == `go test` _after preparing test db_ (just use the user-provided args as is, which is none)
   * or `buffalo test` == `go test ./...` as a special exception (add `./...` for when there is no args at all)

So make `buffalo test [...]` equivalent to `go test [...]` except for the database preparing part, it could be easy to understand the behavior since all Buffalo users are Go developers, and it is more expectable.